portfolio/src/main/java/com/google/sps/servlets/MapServlet.java

```diff
@@ -14,18 +14,58 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/maps")
+@WebServlet("/map")
 public class MapServlet extends HttpServlet {
+  private static final String NAME = "name";
+  private static final String PHONE = "phone";
+  private static final String LOCATION = "location";
+  private static final String DESCRIPTION = "description";
+  private static final String TABLE_NAME = "Pins";
+  private static final String TIME_STAMP = "timestamp";
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("IN DO POST----------------------------------");
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String name = request.getParameter(NAME);
+    String phone = request.getParameter(PHONE);
+    String location = request.getParameter(LOCATION);
+    String description = request.getParameter(DESCRIPTION);
+    long timeStamp = System.currentTimeMillis();
+    System.out.println(name);
+    System.out.println(phone);
+    System.out.println(location);
+    System.out.println(description);
+
+    Entity pinEntity = new Entity(TABLE_NAME);
+    pinEntity.setProperty("Name", name);
+    pinEntity.setProperty("Phone", phone);
+    pinEntity.setProperty("Address", location);
+    pinEntity.setProperty("Description", description);
+    pinEntity.setProperty("TimeStamp", timeStamp);
+    datastore.put(pinEntity);
+
+    response.setContentType("text/html");
+    response.getWriter().println("TEST");
```

Comment:
Turn this into production code?

Reply:
I fixed all of this in the next PR, didn't realize it was still in this one.

Reply:
So do you want to merge these two PRs or pull the fix from next PR to this one?

Reply:
Whichever works, I can merge my latest PR here and then merge this to master. The only thing I want to highlight is that the other PR implements another feature that's slightly separate from this one.

Reply:
I get it. But as we discussed before, each PR should be of high quality by itself.

Reply:
I can't approve this PR as it is. All the review comments are addressed in following PR. I would suggest you either merge this two PRs or pull the comment-addressing code into this PR. Either way, this PR need to be good by itself.

Reply:
I agree with Tong on this one. If you're going to fix the issues in a future PR, then you should at least add a TODO here. Since this is only a couple small comments though it'd probably be best to just recreate the fixes in this branch.